livekit-agents/livekit/agents/beta/tools/http/send_http.py

```diff
@@ -0,0 +1,306 @@
+
+                return response
+
+        except asyncio.TimeoutError as e:
+            logger.warning(
+                "HTTP request timeout",
+                extra={
+                    "url": url,
+                    "method": method,
+                    "timeout_ms": timeout_ms,
+                },
+            )
+            raise ToolError(f"Request timeout after {timeout_ms}ms") from e
+        except aiohttp.ClientError as e:
+            logger.error(
+                "HTTP request client error",
+                extra={
+                    "url": url,
+                    "method": method,
+                    "error": str(e),
+                },
+            )
+            raise ToolError(f"HTTP request failed: {str(e)}") from e
+        except Exception as e:
+            logger.exception(
+                "unexpected error during HTTP request",
+                extra={
+                    "url": url,
+                    "method": method,
+                },
+            )
+            raise ToolError(f"Unexpected error: {str(e)}") from e
+
+
+def create_http_tool(config: HTTPToolConfig) -> FunctionTool:
+    """Create an HTTP request tool.
+
+    The tool makes HTTP requests to a specified endpoint with configurable
+    parameters, headers, and response handling.
+
+    Args:
+        config: Configuration for the HTTP tool
+
+    Returns:
+        FunctionTool that can be used in Agent.tools
+
+    Raises:
+        ValueError: If config is invalid (missing URL, invalid parameters schema)
+
+    Example:
+        ```python
+        from livekit.agents.beta.tools.http import create_http_tool, HTTPToolConfig
+
+        # Minimal usage
+        tool = create_http_tool(HTTPToolConfig(
+            url="https://api.example.com/status"
+        ))
+
+        # With parameters
+        config = HTTPToolConfig(
+            name="create_ticket",
+            description="Create a support ticket",
+            url="https://api.example.com/tickets",
+            method="POST",
+            headers={"Authorization": "Bearer token"},
+            parameters={
+                "type": "object",
+                "properties": {
+                    "title": {"type": "string", "description": "Ticket title"},
+                    "priority": {"type": "string", "enum": ["low", "high"]}
+                },
+                "required": ["title"]
+            }
+        )
+        tool = create_http_tool(config)
+
+        # With normalizer for custom response handling
+        config_with_norm = HTTPToolConfig(
+            name="get_weather",
+            url="https://api.weather.com/current",
+            method="GET",
+            parameters={
+                "type": "object",
+                "properties": {
+                    "city": {"type": "string"}
+                },
+                "required": ["city"]
+            },
+            output_normalizer=lambda resp: f"Temp: {resp.body}" if resp.success else None
```

Comment:
this syntax seems pretty verbose - i wonder if there's a more pythonic way to do it?